### PR TITLE
Add an allow list for RubyGems PATH dependencies

### PR DESCRIPTION
**Pull request by tichavskym**
Signed-off-by: Milan Tichavský <mtichavs@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] New code has type annotations
- n/a OpenAPI schema is updated (if applicable)
- n/a DB schema change has corresponding DB migration (if applicable)
- n/a README updated (if worker configuration changed, or if applicable)

### CODE REVIEW

File: tests/test_workers/test_pkg_managers/test_rubygems.py [hunk 1]
```diff
@@ -603,13 +603,14 @@ def test_download_dependencies(
...
         mock_get_path_info.return_value = path_info
+        allowed_path_deps = {path_info["name"]}
 
         request_id = 1
         # </setup>
 
         # <exercise>
         downloads = rubygems.download_dependencies(
-            request_id, dependencies, mock_bundle_dir.source_root_dir
+            request_id, dependencies, mock_bundle_dir.source_root_dir, allowed_path_deps
```
Comment (chmeliik) [inline tests/test_workers/test_pkg_managers/test_rubygems.py:613]: could we also have a unit test where the PATH dep is not allowed?

File: cachito/workers/config.py [hunk 2]
```diff
@@ -54,6 +54,7 @@ class Config(object):
     cachito_npm_file_deps_allowlist: Dict[str, List[str]] = {}
     cachito_yarn_file_deps_allowlist: Dict[str, List[str]] = {}
     cachito_gomod_file_deps_allowlist: Dict[str, List[str]] = {}
+    cachito_rubygems_file_deps_allowlist: Dict[str, List[str]] = {}
```
[the remark this answers is not in the document]
Comment (tichavskym) [reply cachito/workers/config.py:57]: I will add it in #701

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by lkolacek — APPROVED** [review]
Looks great :+1:

**Pull request merged**